src/otx/core/model/module/classification.py

```diff
@@ -53,18 +56,32 @@ def __init__(
             metric=metric,
         )
 
-        if metric:
-            sig = inspect.signature(metric)
+    def configure_metric(self) -> None:
+        """Configure the metric."""
+        if isinstance(self.metric, partial):
```

Comment:
```suggestion
    def configure_metric(self) -> None:
        """Configure the metric."""
        num_classes_augmented_params = {
             param.default if name != "num_classes" else self.model.num_classes
             for name, param in inspect.signature(self.metric_callable).parameters.items()
             if name != "kwargs"
        }
        self.metric = self.metric_callable(**num_classes_augmented_params)
        return
```

In addition, can it be the base function? Why do we need this function for every derived class?

Reply:
Some metrics (Dice, HLabelAccuracy) should take the different `param_dict` for successfully instantiation. However, I think we can specify only those cases. I'll move this at the base

Reply:
Moved this function to base and leave the functions at the segmentation, h-label classification which needs the special handling. https://github.com/openvinotoolkit/training_extensions/pull/2964/commits/3331c3032f6daad27eb5088f951abb1133a34d8b